pandas/tests/frame/methods/test_interpolate.py

```diff
@@ -64,7 +64,6 @@ def test_interpolate_inplace(self, frame_or_series, request):
         assert np.shares_memory(orig, obj.values)
         assert orig.squeeze()[1] == 1.5
 
-    # TODO(infer_string) raise proper TypeError in case of string dtype
```

Comment:
This still needs to be done?

Reply:
_pandas/core/arrays/arrow/array.py:2164, in ArrowExtensionArray.interpolate_ raises a _ValueError_

```python
        if not self.dtype._is_numeric:
            raise ValueError("Values must be numeric.")
```

whereas for the legacy NumPy object dtype, this gets caught in the internal block code (_pandas/core/internals/blocks.py_) before dispatch to an array method.

```python
        if self.dtype == _dtype_obj:
            # GH#53631
            name = {1: "Series", 2: "DataFrame"}[self.ndim]
            raise TypeError(f"{name} cannot interpolate with object dtype.")

        copy, refs = self._get_refs_and_copy(inplace)

        # Dispatch to the EA method.
        new_values = self.array_values.interpolate(
```

Now, I assume (maybe wrongly) that we don't want to change the EA code to raise a _TypeError_ or change the error message?

Is the solution to catch the _ValueError_ in the block code and raise the TypeError there?

Reply:
I know of no reason, other than I assumed this was a conscious choice when implemented.

Quickly looking at some of the code a `NotImplementedError(f"interpolate is not implemented for dtype={self.dtype}")` maybe more consistent.